core/src/main/java/com/yahoo/imapnio/async/data/QResyncParameter.java

```diff
@@ -0,0 +1,88 @@
+package com.yahoo.imapnio.async.data;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class models the QRESYNC parameter defined in https://tools.ietf.org/html/rfc7162#page-26.
+ */
+public class QResyncParameter {
+    /** Last known UIDVALIDITY. */
+    private long knownUidValidity;
+
+    /** Last known modification sequence. */
+    private long knownModSeq;
+
+    /** Optional set of known UIDs. */
+    private List<MessageNumberSet> knownUids;
+
+    /** Optional parenthesized list of known sequence ranges and their corresponding UIDs. */
+    private QResyncSeqMatchData qResyncSeqMatchData;
+
+    /**
+     * Constructor.
+     *
+     * @param knownUidValidity last known uidvalidity
+     * @param knownModSeq last known modification sequence
+     * @param knownUids known UIDs
+     * @param qResyncSeqMatchData known message sequence set and their corresponding UID
+     */
+    public QResyncParameter(final long knownUidValidity, final long knownModSeq, @Nullable final List<MessageNumberSet> knownUids,
```

Comment:
Suggestion: Add a constructor which just accepts the first two parameters. Which will be the most common use case.

public QResyncParameter(final long knownUidValidity, final long knownModSeq)